src/main/java/athena/TaskList.java

```diff
@@ -96,23 +115,32 @@ public String getDescription(int index) {
      * @param taskNumber Position of task in task list
      * @return Task deleted
      */
-    public Task deleteTask(int taskNumber) {
-        Task taskToDelete = tasks.get(taskNumber);
-        tasks.remove(taskNumber);
+    public Task deleteTask(int taskNumber) throws IndexOutOfBoundsException {
+        Task taskToDelete=null;
+        int counter = -1;
+        int index=-1;
+        for (Task t : tasks) {
+            counter += 1;
+            if (t.getIndex() == taskNumber) {
+                taskToDelete = tasks.get(taskNumber);
+                index = counter;
+            }
+        }
+        tasks.remove(index);
```

Comment:
Wouldn't the index that we assign be different from its index in the list? `tasks.remove(task)` instead?

Reply:
It deletes the correct thing, but I have to agree that the naming is a bit off